taf/tests/test_updater/conftest.py

```diff
@@ -681,3 +681,11 @@ def set_head_commit(auth_repo: AuthenticationRepository):
         auth_repo.set_last_validated_commit(last_valid_commit)
     else:
         raise ValueError("Failed to retrieve the last valid commit SHA.")
+
+
+def pull_client_repos(auth_repo: AuthenticationRepository, client_dir: Path):
```

Comment:
Something like this, yes. We, however, need to have the flexibility to run pull for just one target repo, all target repos and client's auth repo. Create client's auth repo appending origin auth repo's name to client_dir.  Exactly what we want to do depends on the test. This can be split into multiple setup functions, whatever you think it's better